src/Components/Forecast/Forecast.js

```diff
@@ -1,16 +1,219 @@
-        </script>
-    </div>
-  );
+
+
+var localTime = new Date().toLocaleTimeString();
+var weatherIconsMap = {
+  "01d": "wi-day-sunny",
+  "01n": "wi-night-clear",
+  "02d": "wi-day-cloudy",
+  "02n": "wi-night-cloudy",
+  "03d": "wi-cloud",
+  "03n": "wi-cloud",
+  "04d": "wi-cloudy",
+  "04n": "wi-cloudy",
+  "09d": "wi-showers",
+  "09n": "wi-showers",
+  "10d": "wi-day-hail",
+  "10n": "wi-night-hail",
+  "11d": "wi-thunderstorm",
+  "11n": "wi-thunderstorm",
+  "13d": "wi-snow",
+  "13n": "wi-snow",
+  "50d": "wi-fog",
+  "50n": "wi-fog"
+};
+
+
+
+const Forecast = ({data, results}) => {
+
+
+  let [c, setC] = useState();
+  let [f, setF] = useState();
+  var [isFerenheit, setIsFerenheit] = useState(false);
+  function toFerenheit(){
+    if(!isFerenheit){
+      c.style.opacity = 0.5;
+      f.style.opacity = 1;
+      var degrees = (temp * 1.8) + 32;
+      var degrees_max = (temp_max * 1.8) + 32;
+      var degrees_min = (temp_min * 1.8) + 32;
+    setTemp(Math.round(degrees));
+    setTempMax(Math.round(degrees_max));
+    setTempMin(Math.round(degrees_min));
+    setIsFerenheit(true);
+    }
+    
+    
+  }
+  function toCelsius(){
+    f.style.opacity = 0.5;
+    c.style.opacity = 1;
+    setIsFerenheit(false);
+    setTemp(Math.round(results.main.feels_like));
+    setTempMax(Math.round(results.main.temp_max));
+    setTempMin(Math.round(results.main.temp_min));
+  }
+
+  if(results !== null ){
+    var weatherIcon = "wi "+weatherIconsMap[results.weather[0].icon];
+  }
+  let currentTime = new Date().toLocaleTimeString();
+  const [time, setTime] = useState(currentTime);
+  
+  function updateTime() {
+    let time = new Date().toLocaleTimeString();
+    setTime(time);
+    
+  }
+  const [temp, setTemp] = useState();
+  const [temp_max, setTempMax] = useState();
+  const [temp_min, setTempMin] = useState();
+
+  function updateTemp(){
+    if(results !== null){
+      if(results.sys.country === "US" || isFerenheit === true){ // set the default metric to Farenheit for US users
+        var degrees = (results.main.feels_like * 1.8) + 32;
+        var degrees_max = (results.main.temp_max * 1.8) + 32;
+        var degrees_min = (results.main.temp_min * 1.8) + 32;
+        setTemp(Math.round(degrees));
+        setTempMax(Math.round(degrees_max));
+        setTempMin(Math.round(degrees_min));
+        setIsFerenheit(true);
+      }else{
+        setTemp(Math.round(results.main.feels_like));
+        setTempMax(Math.round(results.main.temp_max));
+        setTempMin(Math.round(results.main.temp_min));
+      }
+    }
+      
+  }
+  useEffect(()=> {
```

Comment:
you might need to add setIsFerenheit(false) here to reset the isFerenheit state variable every time the page renders. Otherwise, the isFerenheit value will just stay the same as previously